src/components/Popover/index.tsx

```diff
@@ -98,6 +101,9 @@ export default function Popover({
   const [referenceElement, setReferenceElement] = useState<HTMLDivElement | null>(null)
   const [popperElement, setPopperElement] = useState<HTMLDivElement | null>(null)
   const [arrowElement, setArrowElement] = useState<HTMLDivElement | null>(null)
+  // Default delayBeforeShow is undefined (no delay before showing),
+  // so delayShow should initialize to be true.
+  const [delayShow, setDelayShow] = useState(delayBeforeShow ? false : true)
```

Comment:
This is confusing. Instead, can you default `delayBeforeShow = 0` (on L93), and then:

```suggestion
  const [delayShow, setDelayShow] = useState(Boolean(delayBeforeShow))
```

Reply:
This variable name is confusing. Can you rename to `shouldShow` or `shouldDelayShow` so it's more obvious what it is doing?

Reply:
`shouldShow` is fine, +1 on that change

Reply:
sg, the boolean makes sense... done on the var rename too.